etp-backend/src/test/clj/solita/etp/service/sivu_test.clj

```diff
@@ -0,0 +1,53 @@
+(ns solita.etp.service.sivu-test
+  (:require [clojure.test :as t]
+            [solita.etp.service.sivu :as service]
+            [solita.etp.test-system :as ts]))
+
+(t/use-fixtures :each ts/fixture)
+
+(t/deftest find-empty
+  (t/is (empty? (service/find-all-sivut ts/*db* {:rooli 2 :id 1}))))
+
+(t/deftest find-one-after-add
+  (let [whoami {:rooli 2 :id 1}
+        sivu-in {:title "Pääsääntö"
+                 :body "Äläpä tee virheitä"
+                 :published true
+                 :ordinal 4
+                 :parent-id nil}
+        added-sivu-id (:id (service/add-sivu! ts/*db* sivu-in))
+        sivu-out (service/find-sivu ts/*db* whoami added-sivu-id)]
+    (t/is (not (nil? added-sivu-id)))
+    (t/is (not (nil? sivu-out)))
+    (t/is (= (:id    sivu-out) added-sivu-id))
+    (t/is (= (:title sivu-out) (:title sivu-in)))
+    (t/is (= (:body  sivu-out) (:body  sivu-in)))
+    (t/is (:published sivu-out))))
+
+(t/deftest simple-hierarchy
+  (let [whoami {:rooli 2 :id 1}
```

Comment:
testi-data-setin käyttö voisi tehdä itse testit lyhyemmiksi. Mutta vähän makuasia.

Reply:
Katsotaanpa, tähän asiaan ei vielä tullut paljoa perehdyttyä, mutta ehkä olisi syytä.

Reply:
Nyt äskettäin pushatussa versiossa on generoitua testidataa